cg/services/run_devices/pacbio/run_validator/pacbio_run_validator.py

```diff
@@ -33,11 +36,19 @@ def __init__(
     def ensure_post_processing_can_start(self, run_data: PacBioRunData) -> None:
         """
         Ensure that a post-processing run can start.
-        1. Check if all files are present listed in a manifest file.
-        2. Decompresses the zipped reports.
-        3. Touches a file to indicate that the run is validated
-        4. Skips validation if the run is already validated
+        1. Checks that the post-processing has not been done yet.
+        2. Check if all files are present listed in a manifest file.
+        3. Decompresses the zipped reports.
+        4. Touches a file to indicate that the run is validated
+        5. Skips validation if the run is already validated
+
+        Raises:
+            PostProcessingError: If the run has already been post-processed
         """
+        if Path(run_data.full_path, POST_PROCESSING_COMPLETED).exists():
+            message: str = f"Run {run_data.full_path} is already post-processed, skipping"
+            LOG.warning(message)
+            raise PostProcessingError(message)
```

Comment:
I don't think this should raise an error. merely skipping to the next (i.e. returning) should be enough. An error should happen when something goes wrong. Here we just want nothing to happen.

Reply:
Right now, if it returns None, the service will start the post-processing and that is what we don't want. We want this to stop here. The error is caught anyway

Reply:
Maybe we can add the check to the `_get_all_runs` methods instead? And rename it to `_get_all_runs_to_postprocess` or something?

Reply:
An error does seem a bit weird to raise if this is to be expected

Reply:
or we can make it return a bool, renaming to is_post_processing_ready

Reply:
I think the check is not in the proper place. Ensure_post_processing_can_start has should check if it can start and not if it has already been processed. I think this goes against SRP.

Reply:
Agree, moving to the post-processing service